File: baekjoon/1002/1002.py
'''
터렛 python 풀이
https://doctcoder.tistory.com/entry/%ED%84%B0%EB%A0%9B-%ED%92%80%EC%9D%B4
'''
t=int(input())
for i in range(t):
	x1,y1,r1,x2,y2,r2=map(int,input().split())
	dis=((x1-x2)**2+(y1-y2)**2)**0.5
	rsum=r1+r2
	rsub= r1-r2 if r1>r2 else r2-r1
	cnt=0
	if dis!=0:
		if rsum==dis: cnt+=1
		elif rsub==dis: cnt+=1
		elif rsum<dis or dis<rsub: cnt+=0
		else: cnt+=2
	else:
		if r1==r2: 
			print(-1)
			continue
	print(cnt)

File: baekjoon/1002/1002.cpp
/*
터렛 C++ 풀이
https://doctcoder.tistory.com/entry/%ED%84%B0%EB%A0%9B-%ED%92%80%EC%9D%B4
*/
#include <iostream>
#include <cmath>

using namespace std;

int main()
{
  int t;
	cin>>t;
  while(t!=0)
  {
    int count=0;
    int x1,y1,r1,x2,y2,r2;
    cin>>x1>>y1>>r1>>x2>>y2>>r2;
    double dis=sqrt(pow(x1-x2,2)+pow((y1-y2),2));
    int rsum=r1+r2;	
    int rsub=(r1>r2)?r1-r2:r2-r1;	
    if(dis!=0)	// 
    {
      if(rsum==dis) count++;
      else if (rsub==dis) count++;
      else if (rsum<dis) count+=0;
      else if (dis<rsub) count+=0;
      else count+=2;
    }
    else
    {
      if(r1==r2)
      {
				cout<<"-1\n";
        t--;
        continue;
      }
    }
		cout<<count<<"\n";
    t--;
  }
}
